src/hooks/sounds.tsx

```diff
@@ -49,13 +51,15 @@ export const CustomSoundsProvider: React.FC = ({ children }) => {
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
+  const { user } = useUser();
 
   // Fetch the sound configs data on first load
   useEffect(() => {
+    if (!user?.uid) return;
     fetchSoundConfigs().then((soundConfigs) => {
       setSoundConfigs(soundConfigs);
     });
```

Comment:
```suggestion
  const { user } = useUser();
  const userId = user?.uid;

  // Fetch the sound configs data on first load
  useEffect(() => {
    if (!userId) return;
    
    fetchSoundConfigs().then((soundConfigs) => {
      setSoundConfigs(soundConfigs);
    });
```

Reply:
this way we will prevent useEffect from firing when there is change unrelated to id